packages/mw-error-reporting/src/index.js

```diff
@@ -0,0 +1,21 @@
+import get from 'lodash/get';
+import includes from 'lodash/includes';
+import map from 'lodash/map';
+import forEach from 'lodash/forEach';
+import Promise from 'bluebird';
+
+function errorReportingMiddleware({ reporters, ignoredErrorTypes }) {
+  return function (req, next) {
+    return next()
+        .catch((err) => {
+          if (includes(map(ignoredErrorTypes, 'name'), get(err, 'constructor.name'))) {
+            return Promise.reject(err);
+          }
+
+          forEach(reporters, item => item(err, req));
```

Comment:
several problems here:
1. if `item` (which is a reporter) returns `false`. it will stop the reporters' iteration.
2. if a sync reporter fails, it will fail the entire middleware (no try-catch)
3. if an async reporter fails, you would not know

@sloops77 @udiudi WDYT

Reply:
Agree these are all problems. Here's some solutions:
* rename item => reporter (solves 1 part 1)
* use Promse.map (solves 1 part 2)
* and use .catch(solves 2 & 3)